spec/lib/jobs/dell-racadm-set-bios-job-spec.js

```diff
@@ -0,0 +1,194 @@
+
+        afterEach(function() {
+            encryption.stop();
+            this.sandbox.restore();
+        });
+
+        it("should decrypt a password", function() {
+            var iv = 'vNtIgxV4kh0XHSa9ZJxkSg==';
+            var password = encryption.encrypt('password', iv);
+            var data = {
+                password: password
+            };
+            expect(job.revealSecrets(data)).to.deep.equal({
+                password: 'password'
+            });
+        });
+
+        it("should convert a host mac address to an IP", function() {
+            var macToIP = this.sandbox.stub(lookup, 'macAddressToIp');
+            macToIP.resolves(
+                '10.1.1.2'
+            );
+            var data = {
+                host: '7a:c0:7a:c0:be:ef'
+            };
+            return job.lookupHost(data)
+                .then(function(result){
+                    expect(result).to.deep.equal({
+                        host: '10.1.1.2'
+                    });
+                });
+        });
+    });
+
+    describe('DellRacadm Tool Set Bios', function() {
+        var options;
+
+        beforeEach('Dell Racadm Tool Set BIOS Validation', function() {
+            options = {
+                username:"onrack",
+                password: "onrack",
+                filePath:"//1.2.3.4/src/bios.xml"
+            };
+            job = new RacadmSetBiosJob(options, {}, uuid.v4());
+            mockWaterline.nodes.findByIdentifier = function(){};
+
+            this.sandbox = sinon.sandbox.create();
+            this.sandbox.stub(job, '_subscribeActiveTaskExists').resolves();
+            this.sandbox.stub(mockWaterline.nodes,'findByIdentifier');
+        });
+
+        afterEach('Dell Racadm Tool Set BIOS Validation', function() {
+            this.sandbox.restore();
+        });
+
+        it("should call racadmTool.setBiosConfig", function(){
+            var node = {
+                id: 'bc7dab7e8fb7d6abf8e7d6ac',
+                obmSettings: [
+                    {
+                        service: 'ipmi-obm-service',
+                        config: {
+                            host: '1.2.3.4',
+                            user: 'admin',
+                            password: 'password'
+                        }
+                    }
+                ]
+            };
+
+            var setBiosStub = this.sandbox.stub(racadmTool,'setBiosConfig');
+            var cifsInfo = {user: options.username, password: options.password ,
+                filePath: options.filePath};
+            mockWaterline.nodes.findByIdentifier.resolves(node);
+            setBiosStub.resolves(
+                {
+                    jobStatus:{
+                        status:"Completed"
+                    }
+                }
+            );
+            return job.run()
+                .then(function() {
+                    expect(setBiosStub).to.have.been.called;
+                    expect(setBiosStub).to.have.been.calledWith(
```

Comment:
Since you have expected the stub function to be called with some argument, so it's not necessary to expected it is called.